File: Arrays/Medium/53/53.c
int maxSubArray(int* nums, int numsSize){
    int max = nums[0];
    int currMax = 0;
    for(int i = 0; i < numsSize; i++){
        currMax += nums[i];
        if(currMax > max){
            max = currMax;
        }
        if(currMax < 0){
            currMax = 0;
        }
    }
    return max;
}

File: Arrays/Medium/53/53.py
from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max = nums[0]
        currMax = nums[0]

        for i in range(1, len(nums)):
            currMax = nums[i] if nums[i] > currMax + nums[i] else currMax + nums[i]
            max = currMax if currMax > max else max

        return max